### PR TITLE
Auto assign author to PR

### PR DESCRIPTION
**Pull request by war-in**
Closes #

## Introduced changes

Assigning authors to PRs is tough and sometimes we forget about it. That is why I created the workflow :)

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: .github/workflows/auto-author-assign.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+permissions:
+  pull-requests: write
+
+jobs:
+  assign-author:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'software-mansion/starknet.py' }}  # do not run on forks
+    steps:
+      - uses: toshimaru/auto-author-assign@v1.6.2
```
Comment (cptartur) [inline .github/workflows/auto-author-assign.yml:15]: Looks fine I think, what I'd change is fixing `@version` to a specific version of the commit for extra security, [see here how it's done](https://github.com/software-mansion/starknet.py/blob/development/.github/workflows/dependabot.yml#L26)

File: .github/workflows/auto-author-assign.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# .github/workflows/auto-author-assign.yml
+name: Auto Author Assign
+
+on:
+  pull_request:
```
Comment (THenry14) [inline .github/workflows/auto-author-assign.yml:5]: Correct me if I'm wrong, but wouldn't it trigger this workflow at many events? I guess we could run it only for `opened` and maybe `reopened`

File: .github/workflows/auto-author-assign.yml (hunk 1 above)
Comment (THenry14) [reply .github/workflows/auto-author-assign.yml:15]: We should do it for every action, at the very least not official ones

File: .github/workflows/auto-author-assign.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+permissions:
+  pull-requests: write
+
+jobs:
+  assign-author:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'software-mansion/starknet.py' }}  # do not run on forks
+    steps:
+      - uses: toshimaru/auto-author-assign/push@2daaeb2988aef24bf37e636fe733f365c046aba0  # v1.6.2
```
Comment (THenry14) [inline .github/workflows/auto-author-assign.yml:16]:
```suggestion
      - uses: toshimaru/auto-author-assign@2daaeb2988aef24bf37e636fe733f365c046aba0  # v1.6.2
```

Comment (war-in) [reply .github/workflows/auto-author-assign.yml:16]: Right, found it too 😅

**Review by THenry14 — APPROVED** [review]

**Pull request merged**